src/trousse/dataset.py

```diff
@@ -918,77 +918,6 @@ def to_file(self, filename: Union[Path, str], overwrite: bool = False) -> None:
             except KeyError as e:
                 logging.error(f"Exporting data unsuccessful: \n{e}")
 
-    def fillna(
-        self,
-        columns: List[str],
-        value: Any,
-        derived_columns: List[str] = None,
-        inplace: bool = False,
-    ) -> Optional["Dataset"]:
-        """Fill NaN values ``columns`` (single-element list) column with value ``value``.
-
-        By default NaNs are filled in the original columns. To store the result of filling
-        in other columns, ``derived_columns`` parameter has to be set with the name of
-        the corresponding column names.
-
-        Parameters
-        ----------
-        columns : List[str]
-            Name of the column with NaNs to be filled. It must be a single-element list.
-        value : Any
-            Value used to fill the NaNs
-        derived_columns : List[str], optional
-            Name of the column where to store the filling result. Default is None,
-            meaning that NaNs are filled in the original column. If not None, it must be
-            a single-element list.
-        inplace : bool, optional
-            Whether to modify the current Dataset or return a new instance. Default False,
-            meanining that a new instance of Dataset will be returned.
-
-        Returns
-        -------
-        Optional[Dataset]
-            The new Dataset with NaNs filled if ``inplace=False`` or None otherwise.
-
-        Raises
-        ------
-        ValueError
-            If ``columns`` or ``derived_columns`` are not a single-element list.
-        """
-        if len(columns) != 1:
-            raise ValueError(f"Length of columns must be 1, found {len(columns)}")
-
-        if not inplace:
-            dataset = self._dataset_copy
-        else:
-            dataset = self
-
-        if derived_columns:
-            if len(derived_columns) != 1:
-                raise ValueError(
-                    f"Length of derived_columns must be 1, found {len(derived_columns)}"
-                )
-
-            filled_col = dataset.data[columns[0]].fillna(value, inplace=False)
-            dataset._data[derived_columns[0]] = filled_col
-
-        else:
-            dataset._data[columns[0]].fillna(value, inplace=True)
-
-        if not inplace:
-            return dataset
-
-    @property
-    def _dataset_copy(self) -> "Dataset":
```

Comment:
Why removing this? It could be useful to use it in `FillNA().__call__()` instead of direct `copy.deepcopy(dataset)` so that in the future when we improve the `_dataset_copy` method performance (e.g.: by using `pd.DataFrame().copy()` for copying `data` attribute ), also `FillNA.__call__()` can improve. Am I understanding it wrong?

Reply:
It was okay when we needed this property only inside Dataset (it was private), but having it public just not seemed too pythonic.... if we need to do something fancy within the copy (i.e. the Pandas copy you were mentioning) we can implement `__copy__` and `__deepcopy__` that are the methods called by `copy.copy()` and `copy.deepcopy()` respectively. And we would call `copy.deepcopy()` on the instance anyway. What do you think?

Reply:
Good idea! So that in `__copy__` we will use `.copy(deep=False)` while in `__deepcopy__` we may use `.copy(deep=True)`. It makes sense (it is strange that Pandas implemented in a different way... they are not pythonic like us! )